components/claim/components/ldap_to_email.tsx

```diff
@@ -0,0 +1,201 @@
+    const formClass = classNames('form-group', {'has-error': serverError || passwordError || ldapPasswordError || confirmError});
+
+    const passwordPlaceholder = localizeMessage('claim.ldap_to_email.ldapPwd', 'AD/LDAP Password');
+
+    if (showMfa) {
+        return (
+            <LoginMfa
+                loginId={props.email}
+                password={password}
+                title={{id: t('claim.ldap_to_email.title'), defaultMessage: 'Switch AD/LDAP Account to Email/Password'}}
+                onSubmit={submit}
+            />
+        );
+    }
+    return (
+        <>
+            <h3>
+                <FormattedMessage
+                    id='claim.ldap_to_email.title'
+                    defaultMessage='Switch AD/LDAP Account to Email/Password'
+                />
+            </h3>
+            <form
+                onSubmit={preSubmit}
+                className={formClass}
+            >
+                <p>
+                    <FormattedMessage
+                        id='claim.ldap_to_email.email'
+                        defaultMessage='After switching your authentication method, you will use {email} to login. Your AD/LDAP credentials will no longer allow access to Mattermost.'
+                        values={{
+                            email: props.email,
+                        }}
+                    />
+                </p>
+                <p>
+                    <FormattedMessage
+                        id='claim.ldap_to_email.enterLdapPwd'
+                        defaultMessage='{ldapPassword}:'
+                        values={{
+                            ldapPassword: passwordPlaceholder,
+                        }}
+                    />
+                </p>
+                <div className={formClass}>
+                    <input
+                        type='password'
+                        className='form-control'
+                        name='ldapPassword'
+                        ref={ldapPasswordInput}
+                        placeholder={passwordPlaceholder}
+                        spellCheck='false'
+                    />
+                </div>
+                {ldapPasswordErrorElement}
+                <p>
+                    <FormattedMessage
+                        id='claim.ldap_to_email.enterPwd'
+                        defaultMessage='New email login password:'
+                    />
+                </p>
+                <div className={formClass}>
+                    <LocalizedInput
+                        type='password'
+                        className='form-control'
+                        name='password'
+                        ref={passwordInput}
+                        placeholder={{id: t('claim.ldap_to_email.pwd'), defaultMessage: 'Password'}}
+                        spellCheck='false'
+                    />
+                </div>
+                {passwordErrorElement}
+                <div className={formClass}>
```

Comment:
Only the top form element is supposed to used formClass. These other ones should retain their case specific class names so that they can have specific errors. Otherwise, any error results in all elements showing the error class. Even though I like the change to all the elements being declared as ternaries as terser code, I'd recomment changing back to all the if statements so that we can assign the element *and* assign the input level class names.

This same suggestion applies to `email_to_ldap.tsx`, which has similar code.

Reply:
changes are done for both components